src/renderers/WebGLRenderer.js

```diff
@@ -58,6 +58,11 @@ function WebGLRenderer( parameters ) {
 	let currentRenderList = null;
 	let currentRenderState = null;
 
+	// render() can be called from within a callback triggered by another render.
+	// We track this so that the nested render call gets its state isolated from the parent render call.
+
+	const currentRenderCallStack = [];
```

Comment:
I guess I would call this variable `renderStateStack` since it actually holds render states.

Reply:
Done! 👍